lib/src/https.rs

```diff
@@ -861,13 +854,10 @@ impl HttpsProxy {
         let len = self.listeners.len();
 
         self.listeners
-            .retain(|_, listener| listener.borrow().address.to_string() != remove.address);
+            .retain(|_, listener| listener.borrow().address != remove.address.clone().into());
```

Comment:
Is this clone necessary?

Reply:
TL;DR: yes

We looked into Into/From, they takes ownership no matter what. Implementing them on a reference proves to be tricky.

We think the compiler is probably smart enough to spare the stack's space when "copying" while clonig and reusing memory.

We could implement our own version of From that would not take ownership, but it's 32 bytes of data we're talking about here.
Although a conversion from SocketAddr to the protobuf format will needed at each access log emission, converting to this binary format will still be way more efficient than performing a `to_string()`.
The most crucial performance bottlenecks lie elsewhere, in TLS traffic for instance.

Reply:
I wonder about something like this is feasable?
```suggestion
let remove_address: SocketAddress = remove.address.clone().into();
...
            .retain(|_, listener| listener.borrow().address != remove_address);
```

Reply:
sure can do

Reply:
done!